spark/src/main/java/org/apache/zeppelin/spark/SparkInterpreter.java

```diff
@@ -292,7 +294,20 @@ public SparkContext createSparkContext() {
     }
     if (System.getenv("SPARK_HOME") != null) {
       conf.setSparkHome(System.getenv("SPARK_HOME"));
+
+      // load configs from spark properties file
+      final String sparkPropertiesFile =
+          Utils.getDefaultPropertiesFile(JavaConversions.asScalaMap(System.getenv()));
+      logger.info("Loading spark properties file at " + sparkPropertiesFile);
+
+      final Map<String, String> sparkConfigs =
+          JavaConversions.asJavaMap(Utils.getPropertiesFromFile(sparkPropertiesFile));
+
+      for (Map.Entry<String, String> entry : sparkConfigs.entrySet()) {
+        conf.set(entry.getKey(), entry.getValue());
```

Comment:
I am wondering what was the reason that we don't want to use properties from zeppelin-site.xml

      if (!key.startsWith("spark.") || !val.trim().isEmpty()) {
        logger.debug(String.format("SparkConf: key = [%s], value = [%s]", key, val));
        conf.set(key, val);
      }

The "!" operator doesn't make sense to me. According to the note for this change, it was possibly meant to be if(key.startsWith(...)..). Here was the change note:

ZEPPELIN-46: set only non-empty values for for spark.* properties

Please correct me if I am wrong.